lib/mix/lib/mix/pubsub.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Mix.PubSub do
+  @moduledoc false
+
+  # The Mix pub/sub is responsible for notifying other OS processes
+  # about relevant concurrent events.
+  #
+  # The pub/sub consists of a local subscriber process that receives
+  # events from other OS processes, and a listener supervisor, which
+  # holds all listener processes that have been configured. Whenever
+  # the subscriber receives an event, it sends a message to each of
+  # the listeners.
+  #
+  # Inherently, a compilation may be required before the listener
+  # modules are available, so we start the local subscriber process
+  # separately with `start/0`, and then start the listeners later
+  # with `start_listeners/0`. The subscriber is going to accumulate
+  # events and reply them once the listenres are started.
+
+  @spec start :: :ok
+  def start do
+    case Supervisor.start_child(Mix.Supervisor, Mix.PubSub) do
```

Comment:
It is best to do `Process.whereis(Mix.PubSub)` first, as otherwise we always go through the supervisor, the supervisor then tries to start it, only to get an error back.

Reply:
As you prefer!